Source/engine/render/text_render.cpp

```diff
@@ -686,7 +712,7 @@ uint32_t DrawString(const Surface &out, string_view text, const Rectangle &rect,
 		ClxDraw(clippedOut, characterPosition + Displacement { 0, lineHeight - BaseLineOffset[size] }, sprite);
 	} else if (HasAnyOf(flags, UiFlags::TextCursor) && GetAnimationFrame(2, 500) != 0) {
 		MaybeWrap(characterPosition, 2, rightMargin, initialX, lineHeight);
-		DrawFont(clippedOut, characterPosition, LoadFont(size, color, 0), color, '|', outlined);
+		DrawFont(clippedOut, characterPosition, *LoadFont(size, color, 0), color, '|', outlined);

```

Comment:
Does this call to `LoadFont` always return a non-nullopt value?

Reply:
It can technically fail if the base font is missing from the assets (devilutionx.mpq). It's an edge case we have had for a while as we did not check inside `DrawFont()` either. But I could wrap this in a check to avoid the fail case.

Reply:
Done